addons/hr_expense/models/hr_expense.py

```diff
@@ -822,6 +823,12 @@ def _compute_can_reset(self):
         for sheet in self:
             sheet.can_reset = is_expense_user if is_expense_user else sheet.employee_id.user_id == self.env.user
 
+    def _compute_can_approve(self):
+        is_expense_approver = self.user_has_groups('hr_expense.group_hr_expense_team_approver, hr_expense.group_hr_expense_user')
+        is_expense_manager = self.user_has_groups('hr_expense.group_hr_expense_manager')
+        for sheet in self:
+            sheet.can_approve = is_expense_manager if is_expense_manager else is_expense_approver and sheet.employee_id.user_id != self.env.user
```

Comment:
It could be simplified to something like this:
```suggestion
        can_approve = self.user_has_groups('hr_expense.group_hr_expense_team_approver,hr_expense.group_hr_expense_user,hr_expense.group_hr_expense_manager')
        for sheet in self:
            sheet.can_approve = can_approve and sheet.employee_id.user_id != self.env.user
```

Reply:
Yes! But this way the user who has group_hr_expense_manager rights will not be able to approve his/her own expenses.

Reply:
You are right! I misinterpreted the spec, my bad